domovoi.py

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env python3
+
+import argparse
+import logging
+import os
+import shlex
+import sys
+import time
+
+import settings
+
+from solar_car_process import SolarCarProcess
+
+
+class Domovoi:
+    def start_processes(self, solar_car_processes):
+        for solar_car_process in solar_car_processes:
+            try:
+                solar_car_process.start()
+            except OSError as e:
+                logging.critical(e)
+                self.kill_processes(solar_car_processes)
+                raise
+
+    def kill_processes(self, solar_car_processes):
+        for solar_car_process in solar_car_processes:
+            try:
+                solar_car_process.process.kill()
+            except AttributeError:
+                pass
```

Comment:
Its to ensure that all processes are stopped after a critical error. Let's assume that you have 3 processes p1, p2, and p3 and p1, p3 are running, but p2 isn't able to run. This function stops p1, p3 and ignores the exception of trying to stop p2 because its already stopped.